src/github.com/getlantern/flashlight/app/settings.go

```diff
@@ -136,12 +134,8 @@ func (s *Settings) read() {
 			s.SetSystemProxy(v)
 		}
 
-		if data["userID"] != nil {
-			// This is unmarshaled into a float64, I'm am converting it to string and
-			// then to float32 to catch the case when this is a float32.
-			if id, err := strconv.Atoi(fmt.Sprintf("%v", data["userID"])); err == nil {
-				s.SetUserID(id)
-			}
+		if id, ok := data["userID"].(string); ok {
+			s.SetUserID(id)
```

Comment:
I know this isn't a new thing, but if userID fails to convert, I think we should at least log it if not return an error.